### PR TITLE
New LXC: Fenrus - another personal home page / dashboard

### PR DESCRIPTION
**Pull request by Scorpoon**
## Description
New LXC installation scripts for [Fenrus](https://github.com/revenz/Fenrus) written in C# (.net core 7) and multi-user support.

Installation and update tested with Debian 11/12 and Ubuntu 22.04.
Update process is backing up/restoring the data folder contained in the fenrus publish folder (/opt/fenrus).

For the Homepage:
Default Settings: 
- 512MiB RAM
- 4GB Storage: about ~1.5 GB free after installation
- 1vCPU: building process could be a bit faster when increased but not really much)

Fenrus Dashboard Interface: IP:5000

Credentials are set when the site is opened the first time.
Register is active by default, suggested to be disabled in the settings if not needed.

## Type of change

Please delete options that are not relevant.

- [X] New Script (Develop a new script or set of scripts that are fully functional and thoroughly tested)
- [X] I have performed a self-review of my code, adhering to established codebase patterns and conventions.
- [X] This change requires a documentation update

PS: My first touch with shell scripting, it works without errors but maybe some things could be done better.
I hope it's okay to create a pull request for a new lxc even if it wasn't requested.

### CODE REVIEW

File: install/fenrus-install.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
```
Comment (tteck) [inline install/fenrus-install.sh:20]: $STD apt-get install -y sudo
$STD apt-get install -y mc

File: install/fenrus-install.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  wget -q "https://packages.microsoft.com/config/debian/$var_version/packages-microsoft-prod.deb"
+  $STD dpkg -i packages-microsoft-prod.deb
+  rm packages-microsoft-prod.deb
+fi
+$STD apt-get update
+$STD apt-get install -y dotnet-sdk-7.0
+msg_ok "Installed ASP.NET Core 7 SDK"
+
+msg_info "Installing ${APPLICATION}"
+git clone https://github.com/revenz/Fenrus.git
```
Comment (tteck) [inline install/fenrus-install.sh]: git clone -q https://github.com/revenz/Fenrus.git /opt/${APPLICATION}

File: install/fenrus-install.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+cd Fenrus || exit
```
Comment (tteck) [inline install/fenrus-install.sh]: cd /opt/${APPLICATION}

File: install/fenrus-install.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+dotnet publish -c Release -o "/opt/${APPLICATION}/" Fenrus.csproj
+gitVersionNumber=$(git rev-parse HEAD)
+echo "$gitVersionNumber" >"/opt/${APPLICATION}_version.txt"
+cp -r Apps/ "/opt/${APPLICATION}/"
```
Comment (tteck) [inline install/fenrus-install.sh]: cp -r Apps/ "/opt/${APPLICATION}/" not needed

File: install/fenrus-install.sh [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+cd ..
```
Comment (tteck) [inline install/fenrus-install.sh]: not needed

File: install/fenrus-install.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+rm -r Fenrus/
```
Comment (tteck) [inline install/fenrus-install.sh]: not needed

File: ct/fenrus.sh [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+# Source: https://github.com/revenz/Fenrus
+
+function header_info {
+clear
+cat <<"EOF"
+ _____                         
+|  ___|__ _ __  _ __ _   _ ___ 
+| |_ / _ \ '_ \| '__| | | / __|
+|  _|  __/ | | | |  | |_| \__ \
+|_|  \___|_| |_|_|   \__,_|___/
```
Comment (tteck) [inline ct/fenrus.sh]: https://www.patorjk.com/software/taag/#p=display&f=Slant&t=Fenrus

File: ct/fenrus.sh [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  SD=""
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
```
Comment (tteck) [inline ct/fenrus.sh:57]: I'm sure this will have to be tweaked later. 😉

File: install/fenrus-install.sh (hunk 1 above)
Comment (Scorpoon) [reply install/fenrus-install.sh:20]: Changed it, but could you please explain me why this is needed?

File: install/fenrus-install.sh (hunk 4 above)
Comment (Scorpoon) [reply install/fenrus-install.sh]: Did you try it yourself? Without it the icons for the apps aren't shown and it's also done in the "Dockerfile" on the frenus repo.

File: install/fenrus-install.sh (hunk 1 above)
Comment (tteck) [reply install/fenrus-install.sh:20]: `sudo` is precautionary 

All LXC instances created using this repository come pre-installed with Midnight Commander, which is a command-line tool (`mc`) that offers a user-friendly file and directory management interface for the terminal environment.

Base Dependencies created with all LXCs

Comment (Scorpoon) [reply install/fenrus-install.sh:20]: Ok thank you :)

File: install/fenrus-install.sh (hunk 4 above)
Comment (Scorpoon) [reply install/fenrus-install.sh]: Ah hold on, was to fast. Just saw that you want the raw repo folder and the build output in the same place. I will have a look into it

Comment (tteck) [reply install/fenrus-install.sh]: Screenshot 2024-05-05 8 42 30 AM

File: ct/fenrus.sh (hunk 8 above)
Comment (Scorpoon) [reply ct/fenrus.sh:57]: Yeah i have no long time statistics, we will see

File: install/fenrus-install.sh (hunk 4 above)
Comment (Scorpoon) [reply install/fenrus-install.sh]: Yeah when they are merged it works, you are right ;)

I hope this won't bring up some problems later when the folders are the same.

**Comment by Scorpoon** [comment]
Thanks for reviewing and merging ;)

**Comment by tteck** [comment]
Thank you!

**Pull request merged**